user-operator/src/main/java/io/strimzi/operator/user/model/acl/SimpleAclRule.java

```diff
@@ -118,6 +121,118 @@ public String toString() {
                 "operation: " + operation + ")";
     }
 
+    public AccessControlEntry toKafkaAccessControlEntry(KafkaPrincipal principal)   {
+        AclPermissionType kafkaType;
+        org.apache.kafka.common.acl.AclOperation kafkaOperation;
+
+        switch (type) {
+            case DENY:
+                kafkaType = AclPermissionType.DENY;
+                break;
+            case ALLOW:
+                kafkaType = AclPermissionType.ALLOW;
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid Acl type: " + type);
+        }
+
+        switch (operation) {
+            case READ:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.READ;
+                break;
+            case WRITE:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.WRITE;
+                break;
+            case CREATE:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.CREATE;
+                break;
+            case DELETE:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.DELETE;
+                break;
+            case ALTER:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.ALTER;
+                break;
+            case DESCRIBE:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.DESCRIBE;
+                break;
+            case CLUSTERACTION:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.CLUSTER_ACTION;
+                break;
+            case ALTERCONFIGS:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.ALTER_CONFIGS;
+                break;
+            case DESCRIBECONFIGS:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.DESCRIBE_CONFIGS;
+                break;
+            case IDEMPOTENTWRITE:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.IDEMPOTENT_WRITE;
+                break;
+            case ALL:
+                kafkaOperation = org.apache.kafka.common.acl.AclOperation.ALL;
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid Acl operation: " + operation);
+        }
+
+        return new AccessControlEntry(principal.toString(), getHost(), kafkaOperation, kafkaType);
+    }
+
+    public AccessControlEntryFilter toKafkaAccessControlEntryFilter(KafkaPrincipal principal)   {
+        AclPermissionType kafkaType;
+        org.apache.kafka.common.acl.AclOperation kafkaOperation;
+
+        switch (type) {
+            case DENY:
+                kafkaType = AclPermissionType.DENY;
+                break;
+            case ALLOW:
+                kafkaType = AclPermissionType.ALLOW;
+                break;
+            default:
```

Comment:
I think it would be clearer to have a case for `ANY`, even if it only falls through to the default case.

Reply:
but our current API model doesn't allow `ANY`. The `type` enum, so `AclRuleType` is just `ALLOW` and `DENY`.

Reply:
True, this is going the other way than I thought. Obviously I must be tired :smile: